src/components/Experience.ts

```diff
@@ -47,27 +57,33 @@ export class Experience<TGameStartr extends FullScreenPokemon> extends Component
      */
     public gainExperience(pokemon: IPokemon, experience: number): void {
         pokemon.experience += experience;
-        if (pokemon.experience >= this.gameStarter.equations.experienceStarting(pokemon.title, pokemon.level + 1)
-            && pokemon.statistics.health.current !== 0) {
+        if (pokemon.experience >= this.gameStarter.equations.experienceStarting(pokemon.title, pokemon.level + 1)) {
             this.levelup(pokemon);
         }
     }
 
     /**
      * Calculates how much experience a defeated Pokemon is worth.
-     * https://bulbapedia.bulbagarden.net/wiki/Experience#Gain_formula
      *
      * @param pokemon   The defeated Pokemon.
      * @param trainer   If the defeated Pokemon had a trainer.
+     * @returns   An array of objects of type IExperienceGains.
+     * @see   https://bulbapedia.bulbagarden.net/wiki/Experience#Gain_formula
      */
     public calculateExperience(pokemon: IPokemon, trainer: boolean): IExperienceGains[] {
         const participatedPokemon: IPokemon[] = this.gameStarter.itemsHolder.getItem("battleParticipants");
-        const numerator: number = this.calculateNumerator(pokemon.level, trainer,
-                                                          this.gameStarter.constants.pokemon.byName[pokemon.title.join("")].experience);
+        const numerator: number = this.calculateNumerator(
+            pokemon.level, trainer, this.gameStarter.constants.pokemon.byName[pokemon.title.join("")].experience);
+        const experienceGains: IExperienceGains[] = this.experienceGainBeforeItems(participatedPokemon, numerator);
+        //this.experienceWithItems(experienceGains);
```

Comment:
Then runs the function at line 148, where I'm unsure of how to handle bagactivate